tools/autograd/derivatives.yaml

```diff
@@ -1663,10 +1663,22 @@
   self: to_dense_backward(grad, self)
 
 - name: to_sparse(Tensor self, *, Layout? layout=None, int[2]? blocksize=None) -> Tensor
-  self: grad.to_dense()
+  self: to_sparse_backward(grad, self)
 
 - name: to_sparse.sparse_dim(Tensor self, int sparse_dim) -> Tensor
-  self: grad.to_dense()
+  self: to_sparse_backward(grad, self)
```

Comment:
Since we are using `self` in this formula, when derivatives.yaml is loaded, it will parse the formula and see `self` as a variable we will need save for backward. We should try to avoid that by saving self's layout + blocksize.

Reply:
Agree, now that nested support layout.

Reply:
Hmm alternatively you could add `at::sparse_csr::getBlockSize` as part of replacements:

https://github.com/pytorch/pytorch/blob/d2ee94231e9709e5f4a75f9b6cc83f2869ef3c6f/tools/autograd/load_derivatives.py#L757-L774
And then having the following as formula:
```
self: to_sparse_backward(grad, self.layout(), at::sparse_csr::getBlockSize(self))
```
would automatically save the returned value of getBlockSize instead of self directly.

Reply:
I did not know about that. Thank you, @soulitzer , will do!